src/cluster.c

```diff
@@ -825,6 +825,14 @@ static void updateShardId(clusterNode *node, const char *shard_id) {
     }
 }
 
+static int areInSameShard(clusterNode *node1, clusterNode *node2) {
+    return memcmp(node1->shard_id, node2->shard_id, CLUSTER_NAMELEN) == 0;
+}
+
+static uint64_t nodeEpoch(clusterNode *n) {
+    return (nodeIsSlave(n) && n->slaveof) ? n->slaveof->configEpoch : n->configEpoch;
```

Comment:
n->slaveof should imply nodeIsSlave(n)
```suggestion
    return n->slaveof ? n->slaveof->configEpoch : n->configEpoch;
```

Reply:
I guess this also isn't strictly necessary, but it seems like we copy this around all over the place.

Reply:
yep that is the idea behind single-sourcing. suggestion incorporated.